### DISCUSSION TITLE
Check if vehicle type exists in location types array

### DISCUSSION THREAD

**Discussion by kazitanvirahsan** [google/or-tools#3316] [Routing (and legacy CP) questions]
Hai Team,

I am trying to implement a callback where I can successfully able to compare vehicle_type with a list of the defined types of a location. If any of the types of a location matches the vehicle, the vehicle will be allowed to go to that location. For example,

       locationA_types   = ['red', 'green']
       locationB_types   = ['red', 'green']
       locationC_types   = ['blue']
       locationD_types   = ['blue']

       veh_01_type  =  'red'.     then veh_01 is allowed to go locationA, locationB
       veh_02_type = 'green'   then veh_02 is allowed to go locationA, locationB
       veh_03_type = 'blue'     then veh_03 is allowed to go locationC, locationD

I have vehicle height callback, vehicle weight, and vehicle capacity callback. They all compare one to one. For example, comparing a vehicle's current capacity to load's the weight before pickup. However, I am not really clear on how to compare vehicles' types to a list of types in the callback, and if exists then return true otherwise false. Any idea of how to implement this one-to-many comparison? 

Kazi

**Comment by Mizux** [comment]
to restrict location to a list of vehicles you have few options:

1. use of VehicleVar to limit the list of allowed vehicle
    ```py

    vehicle_red = [0, 1]
    vehicle_green = [2]
    vehicle_blue = [3]
    index = manager.NodeToIndex(42)
    routing.vehicleVar(index).SetValues([-1] + vehicle_red + vehicle_green)
    ```
    ref: https://github.com/google/or-tools/blob/49b6301e1e1e231d654d79b6032e79809868a70e/ortools/constraint_solver/routing.h#L1380-L1382

2. Use a Capacity dimension with vehicle capacity for each colour
    * A location "red" will add +1 in the red capacity unary transit callback
    * vehicle red will have a large capacity of red while others will have 0 i.e. can't visit/load red location
    ref: https://github.com/google/or-tools/blob/49b6301e1e1e231d654d79b6032e79809868a70e/ortools/constraint_solver/routing.h#L550-L553
